references/classification/train.py

```diff
@@ -30,12 +30,12 @@ def train_one_epoch(model, criterion, optimizer, data_loader, device, epoch, arg
     for i, (image, target) in enumerate(metric_logger.log_every(data_loader, args.print_freq, header)):
         start_time = time.time()
         image, target = image.to(device), target.to(device)
-        with torch.cuda.amp.autocast(enabled=args.amp):
+        with torch.cuda.amp.autocast(enabled=scaler is not None):
```

Comment:
Why this change? scaler is not None in case args.amp is true, but having args.amp here makes the code more readable.

Reply:
This is only because in the quantized training script the method `train_one_epoch` is reused. Since there the `amp` param is not supported, accessing `args.amp` breaks the code. I think a more thorough refactoring is needed here but at least this fixes the broken code.